tests/ASN1/ObjectTest.php

```diff
@@ -230,4 +230,98 @@ public function testFromBinaryThrowsException()
         $offset = 10;
         Object::fromBinary($binaryData, $offset);
     }
+
+    /**
+     * @expectedException \FG\ASN1\Exception\ParserException
+     * @expectedExceptionMessage ASN.1 Parser Exception at offset 0: Can not parse binary from data: Offset index larger than input size
+     * @depends testFromBinary
+     */
+    public function testFromBinaryWithEmptyStringThrowsException()
+    {
+        $data = '';
+        Object::fromBinary($data);
+    }
+
+    /**
+     * @expectedException \FG\ASN1\Exception\ParserException
+     * @expectedExceptionMessage ASN.1 Parser Exception at offset 2: Can not parse binary from data: Offset index larger than input size
+     * @depends testFromBinary
+     */
+    public function testFromBinaryWithSpacyStringThrowsException()
+    {
+        $data = '  ';
+        Object::fromBinary($data);
+    }
+
+    /**
+     * @expectedException \FG\ASN1\Exception\ParserException
+     * @expectedExceptionMessage ASN.1 Parser Exception at offset 1: Can not parse content length from data: Offset index larger than input size
+     * @depends testFromBinary
+     */
+    public function testFromBinaryWithNumberStringThrowsException()
+    {
+        $data = '1';
+        Object::fromBinary($data);
+    }
+
+    /**
+     * @expectedException \FG\ASN1\Exception\ParserException
+     * @expectedExceptionMessage ASN.1 Parser Exception at offset 25: Can not parse content length from data: Offset index larger than input size
+     * @depends testFromBinary
+     */
+    public function testFromBinaryWithGarbageStringThrowsException()
+    {
+        $data = 'certainly no asn.1 object';
+        Object::fromBinary($data);
+    }
+
+    /**
+     * @expectedException \FG\ASN1\Exception\ParserException
+     * @expectedExceptionMessage ASN.1 Parser Exception at offset 2: Context-Specific explicitly tagged object [1] starting at offset 2 is shorter than required in the outer tag
+     * @depends testFromBinary
+     */
+    public function testFromBinaryExTagObjWithInvalidOuterLengthThrowsException1()
```

Comment:
Since you test logic from `ExplicitlyTaggedObject` here I suggest we move this into [`ExplicitlyTaggedConstructTest`][1].

While you are at it you could rename that test class to `ExplicitlyTaggedObjectTest` if you feel like it :)

[1]: https://github.com/fgrosse/PHPASN1/blob/master/tests/ASN1/ExplicitlyTaggedConstructTest.php